assets/src/edit-story/components/library/panes/media/media3p/api/apiFetcher.js

```diff
@@ -0,0 +1,126 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import fetch from './fetch';
+
+const API_DOMAIN = 'https://staging-media3p.sandbox.googleapis.com';
+const API_KEY = 'AIzaSyAgauA-izuTeGWFe9d9O0d0id-pV43Y4kE'; // dev key
+
+/**
+ * The methods exposed by the api
+ *
+ * @enum {string}
+ */
+const Paths = {
+  LIST_MEDIA: '/v1/media',
+};
+
+/**
+ * The supported order_by values.
+ *
+ * @enum {string}
+ */
+const OrderBy = {
+  RELEVANCE: 'relevance',
+  LATEST: 'latest',
+};
+
+/**
+ * Perform a GET request to the Media3P API to list media.
+ * If the parameters are invalid or the server does not return a 200 response,
+ * an error is thrown.
+ *
+ * @param {Object} obj - An object with the options for the request.
+ * @param {?string} obj.languageCode The BCP-47 language code, such as "en-US"
+ * or "sr-Latn".
+ * @param {?string} obj.filter  Filter details for items returned.
+ * Filter fields available:
+ * > provider - The media provider to query, or a universal list/search if
+ * left blank.
+ * > type - The media type, either 'image' or 'video'.
+ * > query - A keyword search string.
+ * This may be specified without the 'query:' prefix.
+ * @param {?string} obj.orderBy The sort order for returned results.
+ * Valid sort fields are:
+ * > relevance - The default. This is 'most popular' when listing results,
+ * and 'most relevant' when applying a keyword search.
+ * > latest - Last updated time descending.
+ * @param {?number} obj.pageSize Maximum number of results to be returned by the
+ * server. If unspecified or zero, at most 20 media resources will be returned.
+ * @param {?string} obj.pageToken A page token, received from a previous
+ * `ListMedia` call. Provide this to retrieve the subsequent page.
+ * When paginating, all other parameters provided to `ListMedia` must match
+ * the call that provided the page token.
+ * @return {Promise<Object>} The response from the API.
+ */
+export async function listMedia({
+  languageCode = null,
+  filter = null,
+  orderBy = null,
+  pageSize = null,
+  pageToken = null,
+} = {}) {
+  validateFilter(filter);
+  validateOrderBy(orderBy);
+  validatePageSize(pageSize);
+
+  const params = [
+    ['language_code', languageCode],
+    ['filter', filter],
+    ['order_by', orderBy],
+    ['page_size', pageSize],
+    ['page_token', pageToken],
+    ['key', API_KEY],
+  ].filter((entry) => Boolean(entry[1]));
+  // Querystring always has at least the api key
+  const queryString = params
+    .map((entry) => entry[0] + '=' + entry[1])
```

Comment:
This should to escaped, in case a user enters a search string with a & in it, eg:

const query = Object.keys(params)
    .map(k => k + '=' + encodeURIComponent(params[k]))
    .join('&');

Reply:
Added tests for this.